README.md

```diff
@@ -1,79 +1,345 @@
+5. **Pause System** – ESC to pause, click to resume, with window mode switching (native)
+6. **Respawn System** – Ball respawn after loss with gravity configuration
+7. **Multi-Hit System** – Bricks requiring multiple hits with state transitions
+8. **Texture System** – Asset loading with per-level overrides and fallbacks
 
 ### Plane Constraint
 
-All gameplay bodies lock Y translation (`LockedAxes::TRANSLATION_LOCKED_Y`). Camera sits above looking down, allowing lighting & shadows for 3D feel.
+All gameplay bodies lock Y translation (`LockedAxes::TRANSLATION_LOCKED_Y`). The camera sits above looking down, allowing lighting & shadows for a 3D feel while maintaining 2D gameplay.
 
 ### Collisions
 
-Rapier handles base reflection via restitution. Paddle imparts directional "english" using recent mouse movement. Bricks may later apply custom post-collision effects.
+Rapier handles base reflection via restitution. The paddle imparts directional "english" using recent mouse movement. Bricks may apply custom post-collision effects.
 
 ### Pause Behavior
 
-- **Physics**: Completely frozen when paused (velocities, positions preserved)
-- **Paddle**: Ignores input when paused
-- **Cursor**: Hidden during gameplay, visible when paused
-- **Window Mode** (Native only): Switches fullscreen→windowed on pause, restores on resume
-- **WASM**: Window mode unchanged (WASM doesn't support fullscreen switching)
+| Aspect | Behavior |
+|--------|----------|
+| **Physics** | Completely frozen when paused (velocities, positions preserved) |
+| **Paddle** | Ignores input when paused |
+| **Cursor** | Hidden during gameplay, visible when paused |
+| **Window Mode** (Native) | Switches fullscreen→windowed on pause, restores on resume |
+| **WASM** | Window mode unchanged (WASM doesn't support fullscreen switching) |
 
-## Level File Format
+### Level File Format
 
 Levels live in `assets/levels/` and are RON files parsed into `LevelDefinition`:
 
-```ron
+```rust
 LevelDefinition(
   number: 1,
   gravity: (2.0, 0.0, 0.0), // Optional per-level gravity (x,y,z)
   matrix: [ /* 20 x 20 grid of u8 values */ ]
 )
 ```
 
-### Gravity Override
+#### Gravity Override
+
+If `gravity` is present, it sets `GravityConfig.normal` and `RapierConfiguration.gravity` on load. During paddle growth after respawn, gravity is temporarily set to zero and restored afterward.
+
+#### Matrix Cell Values
+
+| Value | Type | Description |
+|-------|------|-------------|
+| `0` | Empty | No entity |
+| `1` | Paddle | First occurrence only |
+| `2` | Ball | First occurrence only |
+| `3` | Brick | Standard brick |
+| `10-13` | Multi-Hit Brick | Requires multiple hits (13→12→11→10→20→destroyed) |
+| `20` | Simple Brick | Standard destructible brick |
+| `90` | Indestructible Brick | Cannot be destroyed, doesn't count toward completion |
+
+> **Note**: Matrix must be 20×20. Missing paddle or ball results in fallback spawns.
+
+## License
```

Comment:
To be changed to AGPL - this is mainly a learning project, I don't want people to steal it to make money of other people's work